convert2rhel/checks.py

```diff
@@ -16,13 +16,194 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 
+import itertools
 import logging
 import os
+import re
+import subprocess
+
+from convert2rhel.toolopts import tool_opts
+
+
+try:
+    from subprocess import check_output  # pylint: disable=no-name-in-module
+except ImportError:
+    # python2.6 doesn't have this
+    def check_output(command):
+        return subprocess.Popen(
+            command,
+            stdout=subprocess.PIPE,
+        ).communicate()[0]
+
+
+from convert2rhel.systeminfo import system_info
+from convert2rhel.utils import run_subprocess
 
-from convert2rhel.pkghandler import call_yum_cmd
 
 logger = logging.getLogger(__name__)
 
+KERNEL_REPO_RE = re.compile("^.+:(?P<version>.+).el.+$")
+KERNEL_REPO_VER_SPLIT_RE = re.compile("\W+")
+LINK_KMODS_RH_POLICY = "https://access.redhat.com/third-party-software-support"
+
+
+def _convert_to_int_or_zero(s):
+    try:
+        return int(s)
+    except ValueError:
+        return 0
+
+
+def _get_kmod_comparison_key(path):
+    """Create a comparison key from the kernel module abs path.
+
+    Converts /lib/modules/5.8.0-7642-generic/kernel/lib/a.ko.xz ->
+    kernel/lib/a.ko.xz
+
+    Why:
+        all kernel modules lives in /lib/modules/{some kernel release}
+        if we want to make sure that the kernel package is really presented
+        on RHEL, we need to compare the full path, but because kernel release
+        might be different, we compare the relative paths after kernel release.
+    """
+    return "/".join(path.split("/")[4:])
+
+
+def get_host_kmods():
+    try:
+        # TODO make it work with utils.run_subprocess (now fails)
+        kernel_version = check_output(["uname", "-r"]).decode().rstrip()
+        kmod_str = check_output(
+            ["find", "/lib/modules/" + kernel_version, "-name", "*.ko*"]
+        ).decode()
+        assert kmod_str
```

Comment:
```suggestion
        kernel_version, ret_code = utils.run_subprocess("uname -r")
        assert ret_code == 0
        kmod_str, ret_code = utils.run_subprocess("find /lib/modules/" + kernel_version.rstrip() + " -name *.ko*")
        assert kmod_str
        assert ret_code == 0
```

It works for me using the `run_subprocess` on CentOS 6/7/8.

Reply:
Calling `find` is not so pythonic. I'd prefer using for example `glob`, like [here](https://github.com/oamg/convert2rhel/blob/v0.18/convert2rhel/pkghandler.py#L728). But I'll not block the PR on that.